File: server/src/main/java/com/nhom_5/server/service/SeatService.java
package com.nhom_5.server.service;

import com.nhom_5.server.dto.request.BatchSeatTypeRequest;
import com.nhom_5.server.dto.request.SeatRequest;
import com.nhom_5.server.dto.response.SeatResponse;

import java.util.List;
import java.util.UUID;

public interface SeatService {
    List<SeatResponse> getAll();
    List<SeatResponse> getByRoomId(UUID roomId);
    SeatResponse getById(UUID id);
    SeatResponse create(SeatRequest request);
    List<SeatResponse> createRow(com.nhom_5.server.dto.request.CreateRowSeatRequest request);
    SeatResponse update(UUID id, SeatRequest request);
    void updateBatchType(BatchSeatTypeRequest request);
    void delete(UUID id);
    void deleteBatch(com.nhom_5.server.dto.request.BatchSeatDeleteRequest request);
}

File: client/src/services/seatService.ts
import { api } from '@/lib/api'
import type { ApiResponse } from '@/types/api'
import type { Seat, SeatRequest, CreateRowSeatRequest } from '@/types/seat'

export const seatService = {
  getAll: async (): Promise<Seat[]> => {
    const response = await api.get<ApiResponse<Seat[]>>('/seats')
    return response.data || []
  },
  getByRoomId: async (roomId: string): Promise<Seat[]> => {
    const response = await api.get<ApiResponse<Seat[]>>(`/seats/room/${roomId}`)
    return response.data || []
  },

  create: async (payload: SeatRequest): Promise<Seat> => {
    const response = await api.post<ApiResponse<Seat>>('/seats', payload)
    if (!response.data) throw new Error(response.message || 'Tạo ghế thất bại')
    return response.data
  },
  createRow: async (payload: CreateRowSeatRequest): Promise<Seat[]> => {
    const response = await api.post<ApiResponse<Seat[]>>('/seats/batch-row', payload)
    if (!response.data) throw new Error(response.message || 'Tạo hàng ghế thất bại')
    return response.data
  },
  update: async (id: string, payload: SeatRequest): Promise<Seat> => {
    const response = await api.put<ApiResponse<Seat>>(`/seats/${id}`, payload)
    if (!response.data) throw new Error(response.message || 'Cập nhật ghế thất bại')
    return response.data
  },
  updateBatchType: async (seatIds: string[], seatType: Seat['seatType']): Promise<void> => {
    await api.patch<ApiResponse<void>>('/seats/batch-type', { seatIds, seatType })
  },
  delete: async (id: string): Promise<void> => {
    await api.delete<ApiResponse<void>>(`/seats/${id}`)
  },
  deleteBatch: async (seatIds: string[]): Promise<void> => {
    await api.post<ApiResponse<void>>('/seats/batch-delete', { seatIds })
  },
}

